merkletree_test.go

```diff
@@ -0,0 +1,312 @@
+package merkletree
+
+import (
+	"bytes"
+	"crypto/rand"
+	"reflect"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/sha3"
+)
+
+var treeNonce = []byte("TREE NONCE")
+var salt = []byte("salt")
+
+var (
+	pk, sk []byte
+)
+
+func init() {
+	pk, sk, _ = ed25519.GenerateKey(rand.Reader)
```

Comment:
> But I also think it couldn't harm to move the key-generation to specific tests instead of in an init function.

Yes, key generation needs its own test in crypto_test.go, but the need for initialization of a public/private keypair here points to the fact that the Merkle tree constructor probably shouldn't need those, and indeed doesn't really use them. I'll note that above.